password-buffer.c

```diff
@@ -0,0 +1,50 @@
+#include "password-buffer.h"
+#include "log.h"
+#include "swaylock.h"
+#include <stdlib.h>
+
+#ifdef __linux__
+#include <sys/mman.h>
+#endif
+
+char* password_buffer_create(size_t size) {
+	char* buffer = malloc(size);
+	if (!buffer) {
+		swaylock_log_errno(LOG_ERROR, "failed to malloc password buffer");
+		return NULL;
+	}
+
+	if (!password_buffer_lock(buffer, size)) {
+		free(buffer);
+		return NULL;
+	}
+
+	return buffer;
+}
+
+void password_buffer_destroy(char* buffer, size_t size) {
+	clear_buffer(buffer, size);
+	password_buffer_unlock(buffer, size);
+	free(buffer);
+}
+
+bool password_buffer_lock(char* buffer, size_t size) {
+	#ifdef __linux__
+	// Most non-linux platforms require root to mlock()
```

Comment:
I don't think that's the case. The amount of memory a process can `mlock` is in general controlled by `RLIMIT_MEMLOCK`.

Reply:
Indeed. On Linux you can `mlock` up to the configurable RLIMIT_MEMLOCK without privileges, while more than that requires `CAP_IPC_LOCK`. The restrictions are similar on FreeBSD, except if `security.bsd.unprivileged_mlock` is set to 0.

Reply:
I just copy-pasted this. I will delete the ifdef and add some error handling instead